qutebrowser/components/adblock.py

```diff
@@ -131,6 +131,37 @@ def filter_request(self, info: interceptor.Request) -> None:
                         .format(info.request_url.host()))
             info.block()
 
+    def _read_hosts_line(self, line: str) -> list:
+        """Read hosts from the given line.
+
+        Args:
+            line: The line to read.
+
+        Returns:
+            A list containing valid hosts found
+            in the file.
+        """
+        # Remove comments
+        hash_idx = line.find('#')
+        line = line if hash_idx == -1 else line[:hash_idx]
+
+        parts = line.strip().split()
+        if len(parts) == 1:
+            # "one host per line" format
+            hosts = parts[:1]
+        else:
+            # /etc/hosts format
+            hosts = parts[1:]
+
+        accepted_hosts = []
```

Comment:
IMHO that name is confusing, as those are the hosts which end up being blocked, which is the opposite of accepted :wink:

Maybe call it `filtered_hosts`, or just `blocked_hosts`?

Reply:
Also I think this would better be a set, since there's no point in it having duplicates - then this function can also return a set, and the callers can do `target |= self._read_hosts_line(line)` as a set operation.